strikt-core/src/test/kotlin/strikt/Nested.kt

```diff
@@ -47,4 +50,37 @@ internal class Nested {
   ✗ is equal to "bar" : found "foo""""
     )
   }
+
+  @Test
+  fun `nested expectations accepts a suspending lambda`() {
+    expect {
+      that(delayedReturnValue(6)).isEqualTo(6)
+    }
+  }
+
+  @Test
+  fun `all assertions in a block are evaluated even if some fail async`() {
+    fails {
+      expect {
+        that(delayedReturnValue("fnord" as Any?)).isNull()
+        that(delayedReturnValue("FNORD")).isUpperCase()
+        that(delayedReturnValue("FNØRD")).isLowerCase()
+      }
+    }.let { error ->
+      val expected = "▼ Expect that \"fnord\":\n" +
+        "  ✗ is null\n" +
+        "▼ Expect that \"FNORD\":\n" +
+        "  ✓ is upper case\n" +
+        "▼ Expect that \"FNØRD\":\n" +
+        "  ✗ is lower case"
+      assertEquals(expected, error.message)
+    }
+  }
+}
+
+private suspend fun <T> delayedReturnValue(input: T): T {
+  return GlobalScope.async {
+    delay(250)
```

Comment:
Is there any way to test this without introducing a sleep?

Reply:
you can just remove the delay line or change it to `delay(0)`

Reply:
But will that still actually prove anything?

Reply:
yes. it will just not test the coroutines lib. but its just a normal coroutines call, thats not very useful

Reply:
The main thing this is testing is that a the `expect {}` lambda will wait for a suspended function. It doesn't really matter how long it's waiting. I have removed this and it's testing fine.

Reply:
I guess what I'm asking is would it fail without the `suspend` support but I guess there's no way to know as it wouldn't even compile.

Reply:
it tests that it can invoke a suspend method and use its result. adding more code to it would just test the coroutines runtime lib and not strikt. btw for me idea complains about the gradle lock files, but `gradle check` runs fine